vulcanai2/datasets/multidataset.py

```diff
@@ -0,0 +1,121 @@
+# coding=utf-8
+""" Defines the MultiDataset Class"""
+from torch.utils.data import Dataset
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class MultiDataset(Dataset):
+    """
+    Define a dataset for multi input networks.
+
+    Takes in a list of datasets, and whether or not their input_data
+    and target data should be output.
+
+    Parameters
+    ----------
+    datasets : list of tuple(Dataset, use_data_boolean, use_target_boolean)
+        A list of tuples, wherein each tuple should have the Dataset in the
+        zero index, a boolean of whether or not to include the input_data in
+        the first index, and a boolean of whether or not to include the target
+        data in the second index. You can only specificy one target at a time
+        Throughout all incoming datasets.
+
+    Returns
+    -------
+    multi_dataset : torch.utils.data.Dataset
+
+    """
+
+    # TODO: is datasets a reasonable name?
+    # TODO: would be better to make these namedtuples...
+    def __init__(self, datasets):
+        """Initialize a dataset for multi input networks."""
+        def get_total_targets(multi_datasets):
+            num_targets = 0
+            for ds in multi_datasets:
+                if isinstance(ds, MultiDataset):
```

Comment:
shouldn't this be checking for both object types? `MultiDataset` and `Dataset`

Reply:
no because we only need to recurse if it's a multi-dataset. all other Datasets will be accounted for in the else case @priyatharsan

Reply:
try changing the target boolean from the nested one to the top layer multi-dataset in `fashion_....py` and test that it returns the correct target